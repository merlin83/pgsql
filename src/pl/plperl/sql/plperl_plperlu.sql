-- test plperl/plperlu interaction

-- the language and call ordering of this test sequence is useful

CREATE OR REPLACE FUNCTION bar() RETURNS integer AS $$
    #die 'BANG!'; # causes server process to exit(2)
    # alternative - causes server process to exit(255)
    spi_exec_query("invalid sql statement");
$$ language plperl; -- compile plperl code
   
CREATE OR REPLACE FUNCTION foo() RETURNS integer AS $$
    spi_exec_query("SELECT * FROM bar()");
    return 1;
$$ LANGUAGE plperlu; -- compile plperlu code
   
SELECT * FROM bar(); -- throws exception normally (running plperl)
SELECT * FROM foo(); -- used to cause backend crash (after switching to plperlu)


