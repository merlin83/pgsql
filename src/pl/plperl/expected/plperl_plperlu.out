-- test plperl/plperlu interaction
-- the language and call ordering of this test sequence is useful
CREATE OR REPLACE FUNCTION bar() RETURNS integer AS $$
    #die 'BANG!'; # causes server process to exit(2)
    # alternative - causes server process to exit(255)
    spi_exec_query("invalid sql statement");
$$ language plperl; -- compile plperl code
   
CREATE OR REPLACE FUNCTION foo() RETURNS integer AS $$
    spi_exec_query("SELECT * FROM bar()");
    return 1;
$$ LANGUAGE plperlu; -- compile plperlu code
   
SELECT * FROM bar(); -- throws exception normally (running plperl)
ERROR:  syntax error at or near "invalid" at line 4.
CONTEXT:  PL/Perl function "bar"
SELECT * FROM foo(); -- used to cause backend crash (after switching to plperlu)
ERROR:  syntax error at or near "invalid" at line 4. at line 2.
CONTEXT:  PL/Perl function "foo"
